Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -171,55 +171,55 @@ NonnullRefPtr<CSSStyleSheet> Parser::parse_as_stylesheet(TokenStream<T>& tokens)
     return stylesheet;
 }
 
-Vector<Selector> Parser::parse_a_selector()
+NonnullRefPtrVector<Selector> Parser::parse_a_selector()
 {
     return parse_a_selector(m_token_stream);
 }
 
 template<typename T>
-Vector<Selector> Parser::parse_a_selector(TokenStream<T>& tokens)
+NonnullRefPtrVector<Selector> Parser::parse_a_selector(TokenStream<T>& tokens)
 {
     dbgln_if(CSS_PARSER_TRACE, "Parser::parse_a_selector");
 
     auto comma_separated_lists = parse_as_comma_separated_list_of_component_values(tokens);
-    Vector<Selector> selectors;
+    NonnullRefPtrVector<Selector> selectors;
 
     for (auto& selector_parts : comma_separated_lists) {
         auto stream = TokenStream(selector_parts);
         auto selector = parse_single_selector(stream);
-        if (selector.has_value())
-            selectors.append(selector.value());
+        if (selector)
+            selectors.append(selector.release_nonnull());
     }
 
     return selectors;
 }
 
-Vector<Selector> Parser::parse_a_relative_selector()
+NonnullRefPtrVector<Selector> Parser::parse_a_relative_selector()
 {
     return parse_a_relative_selector(m_token_stream);
 }
 
 template<typename T>
-Vector<Selector> Parser::parse_a_relative_selector(TokenStream<T>& tokens)
+NonnullRefPtrVector<Selector> Parser::parse_a_relative_selector(TokenStream<T>& tokens)
 {
     dbgln_if(CSS_PARSER_TRACE, "Parser::parse_a_relative_selector");
 
     auto comma_separated_lists = parse_as_comma_separated_list_of_component_values(tokens);
 
-    Vector<Selector> selectors;
+    NonnullRefPtrVector<Selector> selectors;
 
     for (auto& selector_parts : comma_separated_lists) {
         auto stream = TokenStream(selector_parts);
         auto selector = parse_single_selector(stream, true);
-        if (selector.has_value())
-            selectors.append(selector.value());
+        if (selector)
+            selectors.append(selector.release_nonnull());
```

Comment:
This consumes the selector, is that what you want to do?

Reply:
Not entirely sure what you mean by consumes. That code reads a TokenStream as a Selector and then adds it to the `selectors` list. `selector` and the tokens are not used after that.